src/GitHub/Data/Content.hs

```diff
@@ -87,3 +155,74 @@ instance FromJSON ContentInfo where
+  parseJSON = withObject "ContentResultInfo" $ \o ->
+    ContentResultInfo <$> parseJSON (Object o)
+                  <*> o .: "size"
+
+instance FromJSON ContentResult where
+  parseJSON = withObject "ContentResult" $ \o ->
+    ContentResult <$> o .: "content"
+                  <*> o .: "commit"
+
+instance ToJSON Author where
+  toJSON (Author { authorName   = name
+                 , authorEmail  = email
+                 }) = object
+                 [ "name"      .= name
+                 , "email"     .= email
+                 ]
+
+instance ToJSON CreateFile where
+  toJSON (CreateFile { createFilePath       = path
+                     , createFileMessage    = message
+                     , createFileContent    = content
+                     , createFileBranch     = branch
+                     , createFileAuthor     = author
+                     , createFileCommitter  = committer
+                     }) = object $
+                     [ "path"              .= path
+                     , "message"           .= message
+                     , "content"           .= content
+                     ]
+                     ++ "branch"           .=? branch
+                     ++ "author"           .=? author
+                     ++ "committer"        .=? committer
+
+instance ToJSON UpdateFile where
+  toJSON (UpdateFile { updateFilePath       = path
+                     , updateFileMessage    = message
+                     , updateFileContent    = content
+                     , updateFileSHA        = sha
+                     , updateFileBranch     = branch
+                     , updateFileAuthor     = author
+                     , updateFileCommitter  = committer
+                     }) = object $
+                     [ "path"              .= path
+                     , "message"           .= message
+                     , "content"           .= content
+                     , "sha"               .= sha
+                     ]
+                     ++ "branch"           .=? branch
+                     ++ "author"           .=? author
+                     ++ "committer"        .=? committer
+
+instance ToJSON DeleteFile where
+  toJSON (DeleteFile { deleteFilePath       = path
+                     , deleteFileMessage    = message
+                     , deleteFileSHA        = sha
+                     , deleteFileBranch     = branch
+                     , deleteFileAuthor     = author
+                     , deleteFileCommitter  = committer
+                     }) = object $
+                     [ "path"              .= path
+                     , "message"           .= message
+                     , "sha"               .= sha
+                     ]
+                     ++ "branch"           .=? branch
+                     ++ "author"           .=? author
+                     ++ "committer"        .=? committer
+
+(.=?) :: ToJSON v => KeyValue t => Text -> Maybe v -> [t]
```

Comment:
With exolicit `[Aeson.Pair]` this will be compatible with old aesons, though we can bump lower bump too.